nixos/modules/services/desktops/pipewire/daemon/pipewire-pulse.conf.json

```diff
@@ -37,6 +39,61 @@
       }
     }
   ],
-  "context.exec": [],
-  "stream.properties": {}
+  "context.exec": [
+    {
+      "path": "pactl",
+      "args": "load-module module-always-sink"
+    }
+  ],
```

Comment:
Hi @jansol!

It looks like upstream updated the default pipewire-pulse conf in 0.3.46 to run `pactl load-module module-always-sink` on startup (which dynamically loads a fallback null sink for compatibility like PulseAudio does), but it does not look like `pactl` is in the `PATH` for pipewire-pulse and I get this in my journald logs:

```
$ journalctl --user-unit pipewire-pulse -p err -b
-- Journal begins at Tue 2021-05-25 08:52:13 EDT, ends at Mon 2022-03-14 08:37:47 EDT. --
Mar 14 07:57:04 bina pipewire-pulse[2123]: pw.conf: execvp error 'pactl': No such file or directory
```

It does not really impact my setup all that much (since it's a recoverable error and a new feature I don't use anyway), but it may become an issue for someone. I'm happy to open a PR if you'd like, but I'm not sure the correct way to approach this (e.g. remove this config line, make config have an aboslute path to pactl, add pactl to PATH in pipewire-pulse or systemd user unit, MR to upstream to make pactl path configurable, etc.)


References:

MR for pipewire fallback sink: https://gitlab.freedesktop.org/pipewire/pipewire/-/merge_requests/1159
MR for pulse `module-always-sink`: https://gitlab.freedesktop.org/pipewire/pipewire/-/merge_requests/1161
Specific commit adding `pactl` exec: https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/0b40d36a14861d385a1878760906667a3a65158e

Reply:
Removing the file is out of the question and maintaining a diff just for this seems like an unnecessary maintenance burden.

However I think it would make sense to pull in a basic set of pulseaudio utils when pulseaudio support is enabled. I'm thinking pactl, paplay and whatever tools are pulled into pulseaudio installs by default. Not sure how easy/hard that would be to do though, i.e. how entangled those are in the pulseaudio package.

Reply:
`pipewire-pulse` didn't work on my system until I added 
```
systemd.user.services.pipewire-pulse.path = [ pkgs.pulseaudio ];
```
to my configuration.

I sense a diff may be required :P

Reply:
Pipewire-pulse is working just fine on my machine. It prints the above mentioned error on startup but works as usual after that.

Reply:
Either way, #165125 addresses this.

Reply:
Yeah, sorry, the error turned out to be a different issue caused by stale symlinks.